### PR TITLE
Radio channel

### LINKED ISSUE #724: New channel type GUILD_STAGE_VOICE
A new channel type has been released which should be implemented
```
| GUILD_STAGE_VOICE | 13  | a voice channel for [hosting events with an audience](https://support.discord.com/hc/en-us/articles/1500005513722)
```

### LINKED ISSUE #725: Caching not working due to the new channel type
Due to the new "Radio" channel type, if there is a channel created with the Radio type, Javacord will no longer be able to cache the channel and will throw an `AssertionError` in `ServerImpl` at line 379.

```
java.lang.AssertionError: null
	at java.util.Optional.orElseThrow(Optional.java:401) ~[?:?]
	at org.javacord.core.entity.server.ServerImpl.<init>(ServerImpl.java:379) ~[javacord-core-3.1.2.jar:?]
	at org.javacord.core.util.handler.guild.GuildCreateHandler.handle(GuildCreateHandler.java:33) ~[javacord-core-3.1.2.jar:?]
	at org.javacord.core.util.gateway.PacketHandler.lambda$handlePacket$0(PacketHandler.java:51) ~[javacord-core-3.1.2.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
```

### PR DESCRIPTION
**Pull request by Joshix-1**
should fix #725 

maybe fixes #724, but I don't know if topics are the only thing special about the server stage voice channels.

This is a draft as I still have to look over it tomorrow. I just want to get feedback.

### CODE REVIEW

File: javacord-core/src/main/java/org/javacord/core/util/handler/channel/ChannelCreateHandler.java [hunk 1]
```diff
@@ -118,14 +122,29 @@ private void handleServerVoiceChannel(JsonNode channel) {
...
+
     /**
      * Handles a private channel creation.
      *
      * @param channel The channel data.
      */
     private void handlePrivateChannel(JsonNode channel) {
         // A CHANNEL_CREATE packet is sent every time a bot account receives a message, see
-        // https://github.com/hammerandchisel/discord-api-docs/issues/184
+        // https://github.com/hammerandchisel/discord-api-docs/issues/184Messa
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/util/handler/channel/ChannelCreateHandler.java:147]:
```suggestion
        // https://github.com/hammerandchisel/discord-api-docs/issues/184

File: javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java [hunk 2]
```diff
@@ -337,13 +339,18 @@ public ServerImpl(DiscordApiImpl api, JsonNode data) {
...
+
                 switch (ChannelType.fromId(channel.get("type").asInt())) {
                     case SERVER_TEXT_CHANNEL:
                         getOrCreateServerTextChannel(channel);
                         break;
                     case SERVER_VOICE_CHANNEL:
                         getOrCreateServerVoiceChannel(channel);
                         break;
+                    case SERVER_STAGE_VOICE_CHANNEL:
+                        // TODO Handle them different from voice channel:
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java:351]: Aren't you handling if differently? Or is this a remnant?

File: javacord-api/src/main/java/org/javacord/api/entity/channel/ServerStageVoiceChannel.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.javacord.api.entity.channel;
+
+import org.javacord.api.listener.channel.server.voice.ServerStageVoiceChannelAttachableListenerManager;
+
+import java.util.Optional;
+
+public interface ServerStageVoiceChannel extends ServerVoiceChannel, ServerStageVoiceChannelAttachableListenerManager {
+
```
Comment (Bastian) [inline javacord-api/src/main/java/org/javacord/api/entity/channel/ServerStageVoiceChannel.java:8]:
```suggestion

    @Override
    default ChannelType getType() {
        return ChannelType.SERVER_STAGE_VOICE_CHANNEL;
    }

```

File: javacord-core/src/main/java/org/javacord/core/entity/channel/ServerStageVoiceChannelImpl.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Sets the topic of the channel.
+     *
+     * @param topic The new topic of the channel.
+     */
+    public void setTopic(String topic) {
+        this.topic = topic;
+    }
+
```
Comment (Bastian) [inline javacord-core/src/main/java/org/javacord/core/entity/channel/ServerStageVoiceChannelImpl.java:50]:
```suggestion
    @Override
    public String toString() {
        return String.format("ServerStageVoiceChannel (id: %s, name: %s)", getIdAsString(), getName());
    }

```

**Review by Bastian — CHANGES_REQUESTED** [review]

**Comment by Joshix-1** [comment]
I think as a quick fix for just fixing #725 this is enough.

**Comment by Bastian** [comment]
Lgt :)

Thanks!

**Pull request merged**